bin/magpie-hive-daemon.sh

```diff
@@ -0,0 +1,208 @@
+            $2/bin/metatool -updateLocation ${newfshost} ${line}
+          else
+            echo "Metastore located at ${fshost}, update not required."
+            hiveneedspathupdate=0
+          fi
+        done
+    fi
+
+    if [ -f "${HIVE_PID_DIR}/hive-server.pid" ]
+    then
+        echo "Hive server pid file found, it should already be running"
+    else
+        echo "Starting Hive server on ${myhostname}"
+
+        nohup $2/bin/hive --service hiveserver2 &> /dev/null &
+        scriptpid=$!
+        echo $scriptpid > ${HIVE_PID_DIR}/hive-server.pid
+        sleep 10
+
+        # Check to see if the PATH property in the SERDEPROPERTIES for the tables needs to be updated 
+        if [ $hiveneedspathupdate != "0" ]; then
+            tabledata=`${2}/bin/beeline -u jdbc:hive2://localhost:10000/default -e 'show tables;' |grep '| ' | awk '{print $2}'`
+
+            for table in `echo $tabledata`; do 
+              # this filters out the 'tab_name' header in the sql response
+              if [[ "$table" != "tab_name" ]]; then
+                echo "describe formatted ${table};" >> ${HIVE_LOCAL_SCRATCHSPACE_DIR}/descr_formatted_tables.sql
+              fi
+            done
+
+            # We need to grab the path too, possible that they're located at a different location on a different server?
+            tablepaths=`${2}/bin/beeline -u jdbc:hive2://localhost:10000/default -f ${HIVE_LOCAL_SCRATCHSPACE_DIR}/descr_formatted_tables.sql |grep 'path'`
+            # check for values in tablepaths, it is possible to have no paths to update
+            if [ -n "$tablepaths" ]; then
+              for path in `echo $tablepaths`; do
+                # see if the path variable has hdfs:// in it so no 'junk' ends up in the .sql file
+                if [[ $path = *"hdfs://"* ]]; then
+                  # grab the table name from the end of the path
+                  curtable="${path##*/}"
+                  #grab only the node name from the beginning of the fshost
+                  hivename="${fshost%:*}"
+                  #replace the old node name with the new one in the path 
+                  newpath=$(echo $path | sed -e "s/${hivename}/${myhostname}/g")
+                  echo "Updating OLD: ${path} to NEW: ${newpath}" >> ${HIVE_LOG_DIR}/HIVEDBUPDATELOG.log
+                  echo "Updating OLD: ${path} to NEW: ${newpath}"
+                  echo "ALTER TABLE ${curtable} SET SERDEPROPERTIES ('path'='${newpath}');" >> ${HIVE_LOCAL_SCRATCHSPACE_DIR}/update_hive_path.sql
+                fi
+              done
+              ${2}/bin/beeline -u jdbc:hive2://localhost:10000/default -f ${HIVE_LOCAL_SCRATCHSPACE_DIR}/update_hive_path.sql
+            fi
+
+            #remove both sql files
+            rm ${HIVE_LOCAL_SCRATCHSPACE_DIR}/descr_formatted_tables.sql
+            rm ${HIVE_LOCAL_SCRATCHSPACE_DIR}/update_hive_path.sql
+        fi
+    fi
+else
+  # No nice way to do this, other than to find and kill the Hive and Metastore PIDs
```

Comment:
the tabbing in this "stop" section seems off.  Confused me at first.

Reply:
Sorry...corrected.